homeassistant/components/onkyo/config_flow.py

```diff
@@ -0,0 +1,235 @@
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_ENABLED_SOURCES,
+    CONF_IDENTIFIER,
+    CONF_MAX_VOLUME,
+    CONF_SOURCES,
+    CONNECT_TIMEOUT,
+    DEFAULT_MAX_VOLUME,
+    DEFAULT_SOURCE_NAMES,
+    DEFAULT_SOURCES,
+    DISCOVER_TIMEOUT,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+
+class OnkyoConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Onkyo component."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self._discovered_connections: list[Connection] = []
+        self._connection: Connection | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if not self._discovered_connections:
+            # Discover connections and filter out already configured entries.
+            self._discovered_connections = [
+                conn
+                for conn in await _discover_connections(DISCOVER_TIMEOUT)
+                if conn.identifier
+                not in [
+                    entity.unique_id
+                    for entity in self._async_current_entries(include_ignore=True)
+                ]
+            ]
+
+        if len(self._discovered_connections) == 1:
+            # One connection discovered, immediately connect.
+            self._connection = self._discovered_connections[0]
+            return await self.async_step_connect()
+        if len(self._discovered_connections) > 1:
+            # Multiple connections discovered, let user select.
+            return await self.async_step_select()
+
+        errors["base"] = "discovery_error"
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema({}), errors=errors
+        )
+
+    async def async_step_select(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle multiple network receivers found."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Get selected connection and go to the connect step.
+            self._connection = next(
+                conn
+                for conn in self._discovered_connections
+                if conn.name == user_input["select_receiver"]
+            )
+            return await self.async_step_connect()
+
+        select_scheme = vol.Schema(
+            {
+                vol.Required("select_receiver"): vol.In(
+                    [conn.name for conn in self._discovered_connections]
+                )
+            }
+        )
```

Comment:
So there is no option to add receivers manually... so why not have the integration add all discovered receivers to Home Assistant automatically?

That way, we don't have the ask the user anything? (See for example, Sonos & Cast integrations that do this).

Reply:
I'm now also working on an option to manually add a host, but it requires some changes in the library used.

Reply:
Should we mark this PR draft until that is done?